### PR TITLE
Update doc for continue dev

### PR DESCRIPTION
**Pull request by popcornylu**
[DRC-966](https://linear.app/recce/issue/DRC-966/update-the-doc-site-for-the-state-file)

1. Add content for "Save" and "Import Checklist".
2. Slightly reorganize the state file structure. Since there is significant overlap between the development and PR review scenarios, the common content on saving and using state files has been moved to a shared documentation section. Scenario-specific details for development and PR review are now organized under their respective Scenario: Development and Scenario: PR Review sections.

### CODE REVIEW

File: docs/docs/features/state-file.md [hunk 1]
```diff
@@ -5,52 +5,89 @@ icon: material/file
...
-- [Development](#development): During development, we often need to save the state and even switch between different branches for development.
+- **Checks**: Data from the checks added to the checklist on the Checklist page.
+- **Runs**: Each diff execution in Recce corresponds to a run, similar to a query in a data warehouse. Typically, a single run submits a series of queries to the warehouse and retrieves the final results.
+- **Environment Artifacts**: Includes `manifest.json` and `catalog.json` files for both the base and current environments.
+- **Runtime Information**: Metadata such as Git branch details and pull request (PR) information from the CI runner.
 
-## PR Review
+## How to Save the State File
 
-In the PR review process, the state file can serve as a medium of communication between the submitter and the reviewer.
```
Comment (ijac13) [inline docs/docs/features/state-file.md:15]: we use "author" in our articles.

Comment (ijac13) [reply docs/docs/features/state-file.md:15]: oh, we currently use "submitter" in this doc. We can keep it now and then discuss if we want to align the language.

File: docs/docs/features/state-file.md [hunk 2]
```diff
@@ -5,52 +5,89 @@ icon: material/file
 
 ## Introduction
 
-The state file is the serialized state of a recce instance. There are two scenarios where we need to export the state file:
+The state file represents the serialized state of a Recce instance. It is a JSON-formatted file containing the following information:
 
-- [PR Review](#pr-review): We can include the state file in the PR Review Comment. If the reviewer wants to further connect to the PR environment, they can use this file to see the final results and perform deeper audits.
-- [Development](#development): During development, we often need to save the state and even switch between different branches for development.
+- **Checks**: Data from the checks added to the checklist on the Checklist page.
+- **Runs**: Each diff execution in Recce corresponds to a run, similar to a query in a data warehouse. Typically, a single run submits a series of queries to the warehouse and retrieves the final results.
```
Comment (ijac13) [inline docs/docs/features/state-file.md:11]: Dave and I have a discussion on this. We should call it "Checks" before it is added to Checklist.

Comment (popcornylu) [reply docs/docs/features/state-file.md:11]: Do you have preferred wording? How about

- **Checks**:Data related to checks on the **Checklist** page.

File: docs/docs/features/state-file.md (hunk 1 above)
Comment (popcornylu) [reply docs/docs/features/state-file.md:15]: Ok

File: docs/docs/features/state-file.md (hunk 2 above)
Comment (ijac13) [reply docs/docs/features/state-file.md:11]: ok. here is to follow the data structure in the state file json, so we need to keep run and check...etc. as the state file.

**Pull request merged**